Source/WebCore/css/CSSSelector.cpp

```diff
@@ -336,33 +337,70 @@ std::optional<CSSSelector::PseudoElement> CSSSelector::parsePseudoElementName(St
     return *type;
 }
 
+static std::optional<Style::PseudoElementIdentifier> pseudoElementIdentifierFor(CSSSelectorPseudoElement type)
+{
+    auto pseudoId = CSSSelector::pseudoId(type);
+    if (pseudoId == PseudoId::None)
+        return { };
+    return Style::PseudoElementIdentifier { pseudoId };
+}
+
 // FIXME: We should eventually deduplicate this with CSSSelectorParser::consumePseudo() somehow.
-std::optional<PseudoId> CSSSelector::parsePseudoElement(const String& input, const CSSSelectorParserContext& context)
+std::optional<std::optional<Style::PseudoElementIdentifier>> CSSSelector::parsePseudoElement(const String& input, const CSSSelectorParserContext& context)
```

Comment:
`std::optional<std::optional<` is too confusing. The previous version was better.